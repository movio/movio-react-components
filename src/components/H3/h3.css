@import "variables.css";

:root {
  --secondary-color: var(--color-neutral-a);
  --secondary-font-family: var(--font-secondary);
  --h3-font-family: var(--font-primary);
  --h3-font-size: 1rem;
  --h3-color: var(--color-neutral-2);
}

.h3 {
  font-family: var(--h3-font-family);
  font-size: var(--h3-font-size);
  font-weight: 600;
}

.secondary {
  font-family: var(--secondary-font-family);
  color: var(--secondary-color);
}
