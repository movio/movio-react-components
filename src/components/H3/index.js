import React, { PropTypes } from 'react';
import classnames from 'classnames';

import styles from './h3.css';

/**
 * H3 Component
 * @class
 * @version 0.1
 */
const H3 = ({ children, className }) =>
  <h3 className={classnames(className, styles.h3)}>
    {children}
  </h3>;

/**
 * @property {object}     propTypes                   - H3 PropTypes
 * @property {string}     propTypes.className         - Context class name
 * @property {node}       propTypes.children          - Child node(s)
 */
H3.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};

export default H3;
export {
  styles,
};

