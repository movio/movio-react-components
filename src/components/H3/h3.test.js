import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import H3, { styles } from './index';

describe('<H3 />', () => {
  it('should correctly render children, when text', () => {
    const component = <H3>Title</H3>;
    const wrapper = shallow(component);
    expect(wrapper.text()).toEqual('Title');
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('h3-children-text');
  });

  it('should append a className to the default, if provided', () => {
    const component = <H3 className="test-class">Title H3</H3>;
    const wrapper = shallow(component);
    expect(wrapper.prop('className')).toEqual(`test-class ${styles.h3}`);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('h3-custom-classname');
  });

  it('should use the default className from the stylesheet, if not provided', () => {
    const component = <H3>Title H3</H3>;
    const wrapper = shallow(component);
    expect(wrapper.prop('className')).toEqual(styles.h3);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('h3-default-classname');
  });
});
