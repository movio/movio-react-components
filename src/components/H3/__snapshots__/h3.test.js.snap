// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h3-children-text 1`] = `
<h3
  className="h3"
>
  Title
</h3>
`;

exports[`h3-custom-classname 1`] = `
<h3
  className="test-class h3"
>
  Title H3
</h3>
`;

exports[`h3-default-classname 1`] = `
<h3
  className="h3"
>
  Title H3
</h3>
`;
